src/pages/Statistics/SearchBar.tsx

```diff
@@ -0,0 +1,212 @@
+import Button from "../../components/Button";
+import Card from "../../components/Card";
+import HappinessNumber from "../../components/HappinessNumber";
+import Column from "../../components/layout/Column";
+import Row from "../../components/layout/Row";
+import { QueryKeys } from "../../constants";
+import { useApi } from "../../contexts/ApiProvider";
+import { Happiness } from "../../data/models/Happiness";
+import { formatDate, parseYYYYmmddFormat } from "../../utils";
+import SearchResult from "./SearchResult";
+
+const BASE_FRONTEND_URL = import.meta.env.VITE_BASE_URL;
+
+export default function SearchBar() {
+  const [text, setText] = useState("");
+  const [isFocused, setIsFocused] = useState(false);
+
+  const [startValue, setStartValue] = useState(0);
+  const [endValue, setEndValue] = useState(10);
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [filterShowing, setFilterShowing] = useState(false);
+  const [resultsShowing, setResultsShowing] = useState(false);
+  const [selectedEntryIndex, setSelectedEntryIndex] = useState(-1);
+
+  // handle showing and hiding filter and results
+  const handleToggleFilter = () => {
+    if (!filterShowing || resultsShowing) {
+      setResultsShowing(false);
+      setFilterShowing(true);
+    } else {
+      setFilterShowing(false);
+    }
+  };
+  const handleShowResults = () => {
+    setFilterShowing(false);
+    setResultsShowing(true);
+  };
+
+  const { api } = useApi();
+  const { data } = useQuery<Happiness[]>({
+    queryKey: [
+      QueryKeys.FETCH_HAPPINESS,
+      { start: formatDate(parseYYYYmmddFormat(startDate)) },
+      { end: formatDate(parseYYYYmmddFormat(endDate)) },
+      { low: startValue },
+      { high: endValue },
+      { count: 5 },
+      { text: text },
+    ],
+    queryFn: async () => {
+      const query: Record<string, any> = {
+        low: startValue,
+        high: endValue,
+        count: 5,
+        text: text,
+      };
+      if (!isNaN(new Date(startDate).getTime())) {
+        query.start = formatDate(parseYYYYmmddFormat(startDate));
+      }
+      if (!isNaN(new Date(endDate).getTime())) {
+        query.end = formatDate(parseYYYYmmddFormat(endDate));
+      }
+      const res = await api.get<Happiness[]>("/happiness/search", query);
+      return res.data;
+    }
+  });
+
+
+  // Add keyboard listener for pressing enter and navigation between entries
+  useEffect(() => {
+    // handle when user presses enter to make a search or change selected entry
+    // Initialization of this function must be done in use effect to prevent
+    // data from being capturerd in closure
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Enter") {
+        if (resultsShowing && selectedEntryIndex >= 0 && data && selectedEntryIndex < data.length) {
+          window.open(BASE_FRONTEND_URL + `/home?date=${data[selectedEntryIndex].timestamp}`);
```

Comment:
prolly should use relative url, take a look at this https://github.com/jonathanjma/happiness-app-frontend/blob/1391942c6db273c43c7b8e6cc639a61b80c8a270/src/pages/UserGroups/GroupCard.tsx#L39

Reply:
I didn't know this was a thing, thanks!